src/Renderer/LayeredMaterial.js

```diff
@@ -41,124 +43,116 @@ var getColorAtIdUv = function(nbTex) {
+    fillArray(this.pitScale[COLOR],vector) ;
 
     this.Textures[0] = [emptyTexture];
     this.Textures[1] = Array(nbSamplers);
-    this.paramLayers = Array(8);
-    this.paramBLayers = Array(8);
+    var paramLayers = Array(8);
+    this.nbTexturesLayers = Array(8);
 
-    fill(this.Textures[1],emptyTexture);
-    fill(this.paramLayers,vector4);
-    fill(this.paramBLayers,vector2);
+    fillArray(this.Textures[1],emptyTexture);
+    fillArray(paramLayers,vector4);
+    fillArray(this.nbTexturesLayers,0);
 
     // Elevation texture
-    this.uniforms.dTextures_00 = {
-        type: "tv",
-        value: this.Textures[0]
-    };
-
-    // Color texture
-    this.uniforms.dTextures_01 = {
-        type: "tv",
-        value: this.Textures[1]
-    };
-
-    this.uniforms.nbTextures = {
-        type: "iv1",
-        value: this.nbTextures
-    };
-
-    this.uniforms.layerSequence = {
-        type: "iv1",
-        value: [0, 1, 2, 3, 4, 6, 7, 8]
-    };
-
-    this.uniforms.nColorLayer = {
-        type: "i",
-        value: 1
-    };
-
-    // PIT n Textures
-    // Projection
-    // Visible
-    // Opacity
-
-    this.uniforms.paramLayers = {
-        type: "v4v",
-        value: this.paramLayers
-    };
-
-    this.uniforms.paramBLayers = {
-        type: "v2v",
-        value: this.paramBLayers
-    };
-
-    this.uniforms.pitScale_L00 = {
-        type: "v3v",
-        value: this.pitScale[0]
-    };
-    this.uniforms.pitScale_L01 = {
-        type: "v3v",
-        value: this.pitScale[1]
-    };
-    this.uniforms.lightingOn = {
-            type: "i",
-            value: gfxEngine().lightingOn
-        };
-    this.uniforms.lightPosition = {
-        type: "v3",
-        value: new THREE.Vector3(-0.5, 0.0, 1.0)
-    };
+    this.uniforms.dTextures_00 = new THREE.Uniform(this.Textures[0]);
+
+    // Color textures's layer
+    this.uniforms.dTextures_01 = new THREE.Uniform(this.Textures[1]);
+
+    // Visibility layer
+    this.uniforms.visibility = new THREE.Uniform([true,true,true,true,true,true,true,true]);
+
+    // Array : Number total texture by type
+    this.uniforms.nbTextures = new THREE.Uniform(this.nbTextures);
+
+    // Count color layers
+    this.uniforms.nColorLayer = new THREE.Uniform(1);
+
+    // layer setting
+    // offset color texture | Projection | fx | Opacity
+    this.uniforms.paramLayers = new THREE.Uniform(paramLayers);
+
+    // Elevation texture cropping
+    this.uniforms.pitScale_L00 = new THREE.Uniform(this.pitScale[0]);
+
+    // Color texture cropping
+    this.uniforms.pitScale_L01 = new THREE.Uniform(this.pitScale[1]);
+
+    // Switch light
+    this.uniforms.lightingOn = new THREE.Uniform(gfxEngine().lightingOn);
+
+    // Light position
+    this.uniforms.lightPosition = new THREE.Uniform(new THREE.Vector3(-0.5, 0.0, 1.0));
+
     this.setUuid(id || 0);
     this.wireframe = false;
     //this.wireframe = true;
 
-    this.layerIdToIndex = {};
+    this.layers = [];
```

Comment:
Nope.
You're using it like this `this.layers.push(param.idLayer);` so `layersId` is a better name.
And since you're also doing `this.uniforms.nColorLayer.value = this.layers.length;` a better++ name is: `colorLayersId`